cosmwasm/main.go

```diff
@@ -0,0 +1,252 @@
+				// Examples of adding subcommands:
+				Commands: []plugin.Command{
+					{
+						Use:               "add",
+						Short:             "Add a CosmWasm module to the chain",
+						Long:              "This command will install dependencies, create new files, and modify existing files to add a CosmWasm module to the chain.",
+						PlaceCommandUnder: "ignite cosmwasm",
+					},
+				},
+			},
+		},
+	}, nil
+}
+
+func (p *p) Execute(cmd plugin.ExecutedCommand) error {
+
+	// According to the number of declared commands, you may need a switch:
+
+	switch cmd.Use {
+	case "add":
+		return p.handleAddCommand()
+	default:
+		return fmt.Errorf("unknown command: %s", cmd.Use)
+	}
+}
+
+func (p *p) handleAddCommand() error {
+	fmt.Println("Updating app files...")
+
+	err := installDependencies()
+	if err != nil {
+		return err
+	}
+
+	err = createNewFiles(p.chainName)
+	if err != nil {
+		return err
+	}
+
+	//err = modifyExistingFiles()
+	//if err != nil {return err}
+
+	return nil
+}
+
+func (p) ExecuteHookPre(hook plugin.ExecutedHook) error {
+	return nil
+}
+
+func (p) ExecuteHookPost(hook plugin.ExecutedHook) error {
+	return nil
+}
+
+func (p) ExecuteHookCleanUp(hook plugin.ExecutedHook) error {
+	return nil
+}
+
+func installDependencies() error {
+	cmd := exec.Command("go", "get", "github.com/CosmWasm/wasmd/x/wasm@v0.41.0")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err := cmd.Run()
+	if err != nil {
+		return fmt.Errorf("failed to install dependency: %w", err)
+	}
+	return nil
+}
+
+func replaceWordsInFile(filePath, targetWord, replacement string) error {
+	// Read content from the file
+	content, err := os.ReadFile(filePath)
+	if err != nil {
+		return err
+	}
+
+	// Perform the replacement
+	modifiedContent := strings.ReplaceAll(string(content), targetWord, replacement)
+
+	// Write the modified content back to the file
+	err = os.WriteFile(filePath, []byte(modifiedContent), 0o644)
+	if err != nil {
+		return err
+	}
+
+	//fmt.Printf("Replaced planet with %s\n", replacement)
+
+	return nil
+}
+
+func createFile(inputFilename, outputDir, outputFilename string, chainName string) error {
+
+	// Load the embedded template files
+	templatesFS, err := fs.Sub(templates, "wasm-wiring")
+	if err != nil {
+		fmt.Println("Error:", err)
+		return err
+	}
+
+	// Open the embedded input file
+	inputFile, err := templatesFS.Open(inputFilename)
+
+	if err != nil {
+		return err
+	}
+	defer inputFile.Close() // This line schedules the file to be closed when the function returns
+
+	// Read content from the embedded file
+	sourceContent, err := io.ReadAll(inputFile)
+	if err != nil {
+		return err
+	}
+
+	// Construct the output file path
+	outputPath := filepath.Join(outputDir, outputFilename)
+
+	// Write the embedded content to the output file
+	err = os.WriteFile(outputPath, sourceContent, 0o644)
+	if err != nil {
+		return err
+	}
+
+	// Replace chainName in the output file
+	err = replaceWordsInFile(outputPath, "planet", chainName)
```

Comment:
Use Go template with the `.plush` files could be more easy and fast

Reply:
Resolved with this commit: https://github.com/ignite/apps/pull/2/commits/a523a302a4ab51ed9f7418a0a68c59f9cdd0cc92